core/src/main/java/smile/neighbor/KDTree.java

```diff
@@ -373,7 +373,7 @@ private void search(double[] q, Node node, double radius, List<Neighbor<double[]
             search(q, nearer, radius, neighbors);
 
             // now look in further half
-            if (radius >= diff * diff) {
+            if (radius >= Math.abs(diff)) {
```

Comment:
We has no assumption on `abs(diff) < 1` and I don't see anywhere we do that check. It may be true in your specific data. But we cannot simply assume it in general. Thanks!